src/foam/u2/view/ChoiceView.js

```diff
@@ -246,6 +252,10 @@ foam.CLASS({
       for ( var i = 0 ; i < choices.length ; i++ ) {
         if ( foam.util.equals(choices[i][0], data) ) return choices[i];
       }
+      if ( this.dao && ! this.savedDataSet ) {
```

Comment:
Presumably you'd need to do this in findChoiceByText as well?

Reply:
When `<choice>` is updated, its `<postSet>` will update `<text>` whose `<postSet>` will call `<findChoiceByText>` if necessary